### PR TITLE
Make the size of a browse result page configurable.

### PR DESCRIPTION
**Pull request by mwoodiupui**
## Description
Invent a new configuration property to control the number of results per page when displaying browse results.

## Instructions for Reviewers
List of changes in this PR:
* Create a new configuration property `browseBy.pageSize`.
* Define a default value wherever `browseBy` is being configured.
* Replace hard-coded "20" with the value of this property in the browse-by-metadata component.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by artlowel — COMMENTED** [review]
Thanks @mwoodiupui 

The code looks good, and it works as it should. One strange thing though is that you can't use a number that isn't part of the dropdown of page-sizes on the browse page. It will be rounded to the closest match.

I'm not saying you should change that in this PR (e.g. by making those options configurable as well), but it's likely a good idea to make that clear in the comments

File: src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts [hunk 1]
```diff
@@ -54,7 +55,7 @@ export class BrowseByMetadataPageComponent implements OnInit {
   paginationConfig: PaginationComponentOptions = Object.assign(new PaginationComponentOptions(), {
     id: BBM_PAGINATION_ID,
     currentPage: 1,
-    pageSize: 20
+    pageSize: environment.browseBy.pageSize,
```
Comment (artlowel) [inline src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts:58]: It's better to inject the environment using
```typescript
@Inject(APP_CONFIG) protected appConfig: AppConfig
```
in the constructor, and then use `this.appConfig.browseBy.pageSize` later on. That makes it easier for us to mock the environment in tests.

**Review by artlowel — CHANGES_REQUESTED** [review]
Sorry, I know I already approved, but I noticed something wile reviewing the other RPP PR, I missed here:

**Comment by mwoodiupui** [comment]
"Error:   121:3  error  Angular will not invoke the `ngOnInit` lifecycle method within `@Injectable()` classes  @angular-eslint/contextual-lifecycle"

Suggestions?

**Review by artlowel — APPROVED** [review]
Thanks @mwoodiupui!

File: src/config/default-app-config.ts [hunk 2]
```diff
@@ -205,7 +205,11 @@ export class DefaultAppConfig implements AppConfig {
     // Limit for years to display using jumps of five years (current year - fiveYearLimit)
     fiveYearLimit: 30,
     // The absolute lowest year to display in the dropdown (only used when no lowest date can be found for all items)
-    defaultLowerLimit: 1900
+    defaultLowerLimit: 1900,
+    // The number of entries in a paginated browse results list.
+    // Rounded to the nearest size in the list of selectable sizes on the
+    // settings menu.  See pageSizeOptions in 'pagination-component-options.model.ts'.
+    pageSize: 20
```
Comment (tdonohue) [inline src/config/default-app-config.ts:212]: @mwoodiupui : Can we copy this new config over to our `src/config/config.example.yml`?  That way it's shown alongside the other browse options here: https://github.com/DSpace/dspace-angular/blob/main/config/config.example.yml#L170-L176

Beyond this small change, I think this PR looks good!

Comment (mwoodiupui) [reply src/config/default-app-config.ts:212]: Done.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @mwoodiupui !

**Pull request merged**